autopilot/agent.go

```diff
@@ -606,22 +607,51 @@ func (a *Agent) openChans(availableFunds btcutil.Amount, numChans uint32,
 			continue
 		}
 
-		// Track the available funds we have left.
-		if availableFunds < chanSize {
-			chanSize = availableFunds
+		for {
+			// Get the fee estimated we need to open this channel
+			// in addition to the already selected ones.
+			chans := append(chanSizes, chanSize)
+			feeEstimate, err := a.cfg.ChanController.FeeEstimate(
+				chans,
+			)
+			if err != nil {
+				return err
+			}
+
+			// Give us some wiggle room.
+			feeEstimate *= 3
+
+			// If we have enough balance to cover the fee, we can
+			// break.
+			if chanSize+feeEstimate <= availableFunds {
+				break
+			}
+
+			// Otherwise we need to decrease our channels size and
+			// get another estimate.
+			chanSize = availableFunds - feeEstimate
+			log.Debugf("Resizing channel size to %v - %v = %v "+
+				"to account for fees", availableFunds,
+				feeEstimate, chanSize)
```

Comment:
Would add a comment here to make it clear we continue to ensure we retry the fee estimate as it's possible that we can use less inputs due to the `chanSize` being lower.

Reply:
Done: https://github.com/lightningnetwork/lnd/pull/3017/commits/ed7b36bfc86a13b78e91ff28ff4eaef4a9a12d63